besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -258,5 +263,13 @@ private Long readQbftStartBlockConfig(final QbftConfigOptions qbftConfigOptions)
 
       return startBlock;
     }
+
+    private boolean isCheckpointBlockTotalDifficultyGreaterOrEqualThanTTD(
+        final GenesisConfigOptions configOptions) {
+      return configOptions.getTerminalTotalDifficulty().isPresent()
+          && configOptions.getCheckpointOptions().isValid()
+          && (UInt256.fromHexString(configOptions.getCheckpointOptions().getTotalDifficulty().get())
+              .greaterOrEqualThan(configOptions.getTerminalTotalDifficulty().get()));
```

Comment:
Does changing this back to >= mean @jframe's point about the TransitionController being needed on the boundary case become an issue again?
https://github.com/hyperledger/besu/pull/4735#discussion_r1039180239

Maybe we need a special case for TTD = 0.

Reply:
Added a test and another check for that case